IPython/utils/text.py

```diff
@@ -561,7 +561,13 @@ def _vformat(self, format_string, args, kwargs, used_args, recursion_depth):
 
                 # eval the contents of the field for the object
                 # to be formatted
-                obj = eval(field_name, kwargs)
+                try:
+                    obj = eval(field_name, kwargs)
+                except NameError:
+                    if supressExceptions:
+                       obj = field_name
```

Comment:
I think it's technically easy to issue a warning, but I don't know if we want to, because people use things like `$HOME` without any expectation that it should interpolate a Python variable. I guess you could guess intent because that's environment variables in bash, and they're conventionally capitalised, but it doesn't seem like a very reliable guide.